### DISCUSSION TITLE
Issues with the new window setting close-on-delete: false

### DISCUSSION THREAD

**Discussion by tuurep** [davatorium/rofi#1404] [(Possible) Issues] [answered]
Hello, I've been trying out this new config option:

```
window {
    close-on-delete: false;
}
```

The idea is to open `rofi -show window` for the purpose of closing many windows on one go.

It works such that it doesn't exit on delete-entry (<kbd>Shift</kbd> + <kbd>Del</kbd>),
but it's never possible to close many windows.

Some behaviors I can see when having ~10 windows open and trying to close them all:

1. Window closes, but it creates a false duplicate entry for some other window
2. Window closes, but a wrong list entry is removed
3. Window closes, but the error `Window has vanished` appears in place of something else
4. Selected entry is removed, but nothing closes

I can always fully clear the window list, but the majority of the actual windows remain open. Closing and opening rofi again brings up a correctly updated window listing.

I'm using Openbox window manager.

Does the above list give any hint why keeping `window` open when removing entries might cause false entries appearing? Maybe I can test it out more robustly.

Rofi version number 1.7.0-1

**Comment by DaveDavenport** [comment]
I was afraid of this might happen on some window manager  when implementing this feature on request.
Given I did not get any feedback (And it worked on my i3 setup) I've kept it.

This feature  probably has to be redone or removed.

**Reply by DaveDavenport** [reply]
created issue #1406

**Reply by tuurep** [reply]
Alright thank you this is good to know, I didn't know if it works on other WMs. This is personally a big wish for me on `window` mode, but I don't know if it's a popular request.

**Reply by DaveDavenport** [reply]
The more I think about it, I think it is something  problematic and I  need to admit I should have give it more thought before adding it (and decided not to add it). 

* Closing a window might not close the window, but open a new one (save dialog?) 
* It might take a while to close the window

What needs to be done is to 'listen' to xorg for changes in the window list and this does not match the current design of the window list to well.

**Reply by tuurep** [reply]
Okay, I understand if it's not a good fit. I'm constantly thinking about ways to manage lots of floating windows, I'll see if I can figure something out :)

**Comment by tuurep** [comment, marked as answer]
This is now fixed, see these for details:

#1406
4d0eaf14631a0ef429cba002e63acc15fcfebc32
